src/Quereum.sol

```diff
@@ -180,4 +180,99 @@ contract Quereum {
+
+        // If the function is already closed, we don't need
+        // to do anything.
+        if (questions[question_index].status == 2) {
+            return true;
+        }
+
+        // If the function is open, check whether or not
+        // it should be expired.
+        if (questions[question_index].status == 0) {
+
+            if (questions[question_index].expirationTime >= block.timestamp) {
+                questions[question_index].status = 1;
+            }
+
+            else {
+                return false;
+            }
+        }
+
+        // If the function is expired but not closed,
+        // assign rewards.
+        if (questions[question_index].status == 1) {
+
+            uint256 max_endorsements = -1;
+            address[] most_endorsed_authors = [];
+
+            // Traverse through all the answers for this question.
+            for (uint256 response_index = 0;
+                response_index < questions[question_index].responses.length;
+                response_index++) {
+                
+                address_index = questions[question_index].responses[response_index];
+
+                if (answers[answer_index].endorsements > max_endorsements) {
+                    max_endorsements = answers[answer_index].endorsements;
+                    most_endorsed_authors = [answers[answer_index].author];
+                }
+
+                else if (answers[answer_index].endorsements == max_endorsements) {
+                    most_endorsed_authors.push(answers[answer_index].author);
+                }
+            }
+
+            // If max_endorsements is not -1, reward the recipients.
+            if (max_endorsements != -1) {
+                uint256 reward = questions[question_index].reward;
+                uint256 allocation = reward / most_endorsed_authors.length;
+
+                for (uint256 i = 0; i < most_endorsed_authors.length; i++) {
+                    balances[most_endorsed_authors[i]] += allocation;
+                }
+            }
+
+            // If max_endorsements is -1, nobody answered the question.
+            // Give the entire reward back to the original asker.
+            else {
+                uint256 reward = questions[question_index].reward;
+                balances[questions[question_index].user] += reward;
+            }
+
+            questions[question_index].rewardAllocated = true;
+        }
+
+        return true;
+    }
+
+    // This function allows the poster of a question to 
+    // close the question without allocating any rewards.
+    function close_without_reward(uint256 question_index) public {
+        require(msg.sender == questions[question_index].user,
+            "Only the question poster can close the question.");
+        require(!questions[question_index].rewardAllocated,
+            "Rewards have already been allocated.");
+
+        questions[question_index].status = 2;
+        questions[question_index].rewardAllocated = true;

```

Comment:
Technically, the reward hasn't been allocated, but we want to set this boolean to true so that in the future, if someone checks this variable and sees that the reward hasn't been allocated, they don't accidentally allocate the reward, when the original poster intended for the reward to not be distributed.